src/Esfa.Vacancy.Register.Api/App_Start/VacancyApiExceptionHandler.cs

```diff
@@ -23,49 +26,69 @@ public override void Handle(ExceptionHandlerContext context)
-                    context.Result = CreateResult(HttpStatusCode.NotFound, ((ResourceNotFoundException) context.Exception).Message, context.Request);
+                    context.Result = CreateStringResult(HttpStatusCode.NotFound, ((ResourceNotFoundException) context.Exception).Message, context.Request);
                     return;
                 }
 
                 if (context.Exception is InfrastructureException)
                 {
                     logger.Error(context.Exception.InnerException, GetLogMessage("Unexpected infrastructure error", context));
-                    context.Result = CreateResult(HttpStatusCode.InternalServerError, GenericErrorMessage, context.Request);
+                    context.Result = CreateStringResult(HttpStatusCode.InternalServerError, GenericErrorMessage, context.Request);
                     return;
                 }
 
                 logger.Error(context.Exception, GetLogMessage("Unexpected error", context));
-                context.Result = CreateResult(HttpStatusCode.InternalServerError, GenericErrorMessage, context.Request);
+                context.Result = CreateStringResult(HttpStatusCode.InternalServerError, GenericErrorMessage, context.Request);
             }
             catch
             {
                 //whatever happens don't leak the stack trace
-                context.Result = CreateResult(HttpStatusCode.InternalServerError, ExceptionInExceptionHandlerErrorMessage, context.Request);
+                context.Result = CreateStringResult(HttpStatusCode.InternalServerError, ExceptionInExceptionHandlerErrorMessage, context.Request);
             }
         }
 
-        private IHttpActionResult CreateResult(HttpStatusCode statusCode, string content, HttpRequestMessage request)
+        private IHttpActionResult CreateStringResult(HttpStatusCode statusCode, string content, HttpRequestMessage request)
         {
             var response = new HttpResponseMessage(statusCode);
             response.Content = new StringContent(content);
             var result = new CustomErrorResult(request, response);
             return result;
         }
 
+        private IHttpActionResult CreateBadRequestResult(ValidationException validationException, HttpRequestMessage request)
+        {
+            var badrequestResponse = new BadRequestResponse
+            {
+                RequestErrors = validationException.Errors
+                    .Select(validationFailure => new BadRequestError
+                    {
+                        ErrorCode = validationFailure.ErrorCode,
+                        ErrorMessage = validationFailure.ErrorMessage
+                    })
+            };
+
+            var response = new HttpResponseMessage(HttpStatusCode.BadRequest)
+            {
+                Content = new ObjectContent<BadRequestResponse>(badrequestResponse, new JsonMediaTypeFormatter())
```

Comment:
I don't understand why you need to do this exactly. Can we speak about this as I think we shouldn't need to specify how the output is formatted here. Also I think web api handles responses in both xml and json, with json being default but it is dependent on the request.